src/languages/php.js

```diff
@@ -339,6 +340,62 @@ export default function(hljs) {
     }
   };
 
+  // This will not detect camelCase classes
+  const PASCAL_CASE_CLASS_NAME = regex.concat("[A-Z]", IDENT_RE);
+
+  const CONSTANT_REFERENCE = regex.concat(IDENT_RE, "\\b(?!\\()");
```

Comment:
How is this a constant reference?  In other languages our constant rule is ALL_CAPS_CASE.

Reply:
And if it's truly a constant we'd use `variable.constant`...

Reply:
> How is this a constant reference? In other languages our constant rule is ALL_CAPS_CASE. Is anything following a `::` just a constant in PHP?

Language allows also lower case or any other, but probably almost all use ALL_CAPS_CASE.  
It is just IDENT_RE that can't be followed by (. This distinguishes it from the static method call. I left examples in the PR description. In this context, it's just a helper that is part of a few matching rules.


> And if it's truly a constant we'd use `variable.constant`...

Ok, I guess accessing Enum value is also `variable.constant`, because we can't distinguish those.   

Constant declaration and Enum values also should be marked with `variable.constant`?

Reply:
> It is just IDENT_RE that can't be followed by (.

That's that's a "function dispatch" or something, not a "constant"... constant to me means `const` or read only or `UPPERCASE` by convention in may languages...

> Constant declaration and Enum values also should be marked with variable.constant?

Perhaps if you could detect they are enum values... I dunno what the PHP syntax is... for most languages it's impossible to do this 100% correctly so we usually have a UPPERCASE constant rule for languages where it's a string pattern and then only catch others in very explicit cases like `class name`, etc... where it's clean name is constant despite the weird casing.

Reply:
Ah I just looked above, yes the enum declarations could be marked that way I suppose...